raiden/smart_contracts/ChannelManagerLibrary.sol

```diff
@@ -64,63 +35,50 @@ library ChannelManagerLibrary {
         uint settle_timeout)
         returns (address channel_address)
     {
-
         channel_address = new NettingChannelContract(
             self.token,
             caller_address,
             partner,
             settle_timeout
         );
 
-        self.node_channels[caller_address].push(channel_address);
-        self.node_channels[partner].push(channel_address);
-        self.all_channels.push(channel_address);
+        self.channel_addresses[partyHash(caller_address, partner)] = channel_address;
     }
 
     /// @notice Remove a channel after it's been settled
     /// @dev Remove channel after it's been settled
-    /// @param caller_address Τhe address of the caller
-    /// @param partner Αddress of the partner
-    /// @param channel_address Αddress of the channel to be closed
-    /// @param caller_index Τhe index of the caller in our channels
-    /// @param partner_index Index of the partner in partner_channels
+    /// @param caller_address address of the caller
+    /// @param partner of the partner
     function deleteChannel(
         Data storage self,
         address caller_address,
-        address partner,
-        address channel_address,
-        uint caller_index,
-        uint partner_index)
+        address partner)
         internal
     {
-        address[] our_channels = self.node_channels[caller_address];
-        address[] partner_channels = self.node_channels[partner];
-
-        // move last element of array to i_index pos
-        our_channels[caller_index] = our_channels[our_channels.length - 1];
-        our_channels.length--;
-        // move last element of array to j_index pos
-        partner_channels[partner_index] = partner_channels[partner_channels.length - 1];
-        partner_channels.length--;
+        // remove channel from channel_addresses.
+        self.channel_addresses[partyHash(caller_address, partner)] = 0x0;
+    }
 
-        // remove address from all_channels
-        for (uint k = 0; k < self.all_channels.length; ++k) {
-            if (self.all_channels[k] == channel_address) {
-                self.all_channels[k] == self.all_channels[self.all_channels.length - 1];
-                self.all_channels.length--;
-                break;
-            }
+    /// @notice partyHash get the hash of the two addresses
+    /// @dev Get hash of two addresses
+    /// @param address_one address of one party
```

Comment:
Also please add a comment specifying that supplying two equal addresses results in an error.